### PR TITLE
Introduce logrotate

### LINKED ISSUE #653: Ensure Deletion of Logfiles older than Two weeks
In order to be DIPA compliant we need to ensure that logfiles older than 2 weeks are deleted.

### Implementation proposal

There are several ways on how the logging stack of Docker can be configured during cloud init. There may be an option to write to a new file every day and to rotate the log every day for up to fourteen days. In case there is no such option we may have to write a custom logger which provides this option.

### Definition of Done

There shall be a test that checkes that there are no logfiles older than 14 days. As staging is usually updated in shorter periods of time (where logs are being deleted) that test should be run on the canary landscape.

### Side conditions

We will be writing the logs to an external logging stack (EFK). This logging stack must be made compliant as well.

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
- Enables [logrotate](https://linux.die.net/man/8/logrotate) for container log rotation. 
- Configures [journal](https://www.freedesktop.org/software/systemd/man/journald.conf.html) to rotate the system components that do not run in container and log in journal - kubelet, the container runtime (docker). 

**Which issue(s) this PR fixes**:
Fixes #653 and #662.

**Special notes for your reviewer**:
See:
- https://kubernetes.io/docs/concepts/cluster-administration/logging/#logging-at-the-node-level
- https://docs.docker.com/config/containers/logging/json-file/

**Release note**:

```noteworthy operator
All shoot worker machines are now configured with log rotation for user containers and system components. Log files are kept for a maximum of 14 days.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
WIP?

File: charts/shoot-cloud-config/charts/original/templates/docker/_docker.service
Comment (rfranzke) [inline charts/shoot-cloud-config/charts/original/templates/docker/_docker.service]: Can't we delete the whole file?

Comment (ialidzhikov) [reply charts/shoot-cloud-config/charts/original/templates/docker/_docker.service]: Thanks for the spot. I also removed `docker.service` from the unit file - the result is the same as before, docker.service is active (running), but not enabled. 
@rfranzke, @mvladev, @adracus, a small off-topic - shouldn't we also enable the docker.service in the `coreos.units`, unfortunately I cannot find any meaningful info from coreos docs?

Comment (rfranzke) [reply charts/shoot-cloud-config/charts/original/templates/docker/_docker.service]: Not sure, no strong opinion about this one as it does not change the behaviour anyways.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
Waiting for more feedback before merging.

**Review by mvladev — APPROVED** [review]
/lgtm

**Pull request merged**
